src/classes/DRS_ContactMerge_CTRL.cls

```diff
@@ -0,0 +1,1036 @@
+            for(DuplicateRecordSet dupRecSet : duplicateRecordSets) {
+                allDRIIds.addAll(new Map<Id, DuplicateRecordItem>(dupRecSet.DuplicateRecordItems).keySet());
+                allDRIs.addAll(new Map<Id, DuplicateRecordItem>(dupRecSet.DuplicateRecordItems).values());
+            }
+            // Construct query to fetch all the related contact records to find the first contact related to the DRS record
+            Integer queryLimit = 30000;
+            List<String> contactField = new List<String>();
+            contactField.add('Id');
+            contactField.add('Name');
+            contactField.add('createdDate');
+             
+            String whereString = new UTIL_Query()
+                .withFrom('DuplicateRecordItem')
+                .withSelectFields(driFieldNames)
+                .withWhere('Id IN :allDRIIds')
+                .build();
+            whereString = 'Id IN (' + whereString + ')'; 
+            String contactQueryStr =  new UTIL_Query()
+                .withFrom('Contact')
+                .withSelectFields(contactField)
+                .withWhere(whereString)
+                .withLimit(queryLimit)
+                .build();
+            Map<Id,Contact> contacts = new Map<Id,Contact>((List<Contact>)Database.query(contactQueryStr));
+            for(DuplicateRecordItem dri : allDRIs) {
+                Contact con = contacts.get(dri.RecordId);
+                if(!(firstContactOfDRS.containsKey(dri.DuplicateRecordSetId))) {
+                    firstContactOfDRS.put(dri.DuplicateRecordSetId,con);
+                }
+                else {
+                    if(con.createdDate < firstContactOfDRS.get(dri.DuplicateRecordSetId).createdDate) {
+                        firstContactOfDRS.put(dri.DuplicateRecordSetId,con);
+                    }
+                }
+            }
+            totalPages = getTotalPages(); 
+        } catch (exception ex) {
+                ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.Error, ex.getMessage()));
+        }
+    }
+    
+    /*******************************************************************************************************
+    * @description Construct SOQL query to fetch all the contact related Duplicate Record Sets
+    * @return String Returns the query to fetch Duplicate Record Sets
+    */
+    public String constructDRSSOQL() {
+        //build SOQL subquery on Duplicate Record Items
+        String subQueryOrderBy = 'CreatedDate ASC';
+        
+        String subQuery =  new UTIL_Query()
+            .withFrom('DuplicateRecordItems')
+            .withSelectFields(driFieldNames)                                  
+            .build();
+        String subQueryStr = '('+subQuery+')';
+        drsFieldNames.add(subQueryStr);
+        String setOrderBy = 'LastModifiedDate DESC ';
+        String setWhereClause = 'RecordCount > 1 AND DuplicateRule.SObjectType = \'Contact\'';
+        Integer soqlLimit = 7000;
```

Comment:
Why 7000? Could this be a defined constant in the class?

Reply:
@ptewson-sfdo We have added a limit of 7000 to DRS query because we were facing the 50001 query rows issue. As we are displaying the first contact for each DRS record, we had to maintain a map with DRS id and its respective first contact record. To form this map we are querying on the duplicate contacts in the org and if the count of contact records is more we are facing the 50001 query rows issue. Hence to limit the Contacts to be retrieved we limited the associated DRS records to 7000 but as we are bulk testing we might have to lower it down to 5000. Because of this same issue we have added a limit of 30000 to the contact records query which we might have to reduce after bulk testing.